File: packages/forum_forge/tests/components.test.ts
import { describe, it, expect } from 'vitest'
import components from '../seed/components.json'

interface ComponentDefinition {
  id: string
  type: string
  children: unknown[]
}

describe('Forum Forge Components', () => {
  it('should be a valid array', () => {
    expect(components).toBeInstanceOf(Array)
  })

  it('should have valid component structure if components exist', () => {
    if (components.length > 0) {
      components.forEach((component: ComponentDefinition) => {
        expect(component.id).toBeDefined()
        expect(component.type).toBeDefined()
        expect(typeof component.id).toBe('string')
        expect(typeof component.type).toBe('string')
        expect(component.children).toBeInstanceOf(Array)
      })
    }
  })
})

File: packages/forum_forge/seed/scripts/tests/components.test.lua
-- Component validation tests for forum_forge package
-- Uses lua_test framework

describe("Forum Forge Components", function()
  local components = load_cases("components.json")
  
  it("should be a valid array", function()
    expect(components).toBeType("table")
  end)
  
  it("should have valid component structure if components exist", function()
    if #components > 0 then
      for _, component in ipairs(components) do
        expect(component.id).toBeTruthy()
        expect(component.type).toBeTruthy()
        expect(component.id).toBeType("string")
        expect(component.type).toBeType("string")
      end
    end
  end)
end)
